llm_toolkit/prompt_builder.py

```diff
@@ -797,9 +797,11 @@ def _format_constructors(self) -> str:
       constructor_sig = ctr.get('function_signature')
       if constructor_sig:
         constructors.append(f'<signature>{constructor_sig}</signature>')
+        self.benchmark.exceptions.extend(ctr.get('exceptions'))
 
     if constructors:
       ctr_str = '\n'.join(constructors)
+      self.benchmark.exceptions = list(set(self.benchmark.exceptions))
```

Comment:
Is `list(set(` to avoid duplicates? If so why not guarantee the creator of the `benchmark.exceptions`?